### ISSUE TITLE
react-ts adaptor pack: first domain adaptor (training data, weights, eval suite, system prompt)

### ISSUE THREAD

**Issue by falese** [falese/coder#11]
## Overview
Build, train, and publish the react-ts adaptor pack — MUI components + Module Federation patterns. This is a multi-step undertaking that spans data curation, training, eval authoring, and packaging. The central goal is a **measurable, reproducible demonstration** that the adaptor produces better output than the base model alone on React/MUI/MFE tasks.

## Prerequisites (must be done before starting)

- [x] #9 (adaptor eval) complete — needed for `coder adaptor eval react-ts`
- [x] `ts_declare` and `constructor_call` anchors added to `coder data extract` — new anchors not yet implemented (noted in #7 post-close comment)

## Step 1 — Add missing extraction anchors to `coder data extract`

Before any data can be extracted, two new anchors must be implemented in `src/data/extract.ts`:

| New anchor | Side | Matches |
|---|---|---|
| `ts_declare` | prompt | `declare module '...' {` header line |
| `declare_body` | completion | body inside `declare module { }` |
| `constructor_call` | completion | `new ClassName({...})` expression following the prompt |

## Step 2 — Source data

Clone both repos and pin to a specific release tag:

```bash
git clone https://github.com/mui/material-ui --depth 1 --branch v6.4.0
# Target: packages/mui-material/src/**/*.tsx

git clone https://github.com/module-federation/module-federation-examples --depth 1
# Target: **/*.ts, **/*.tsx, webpack.config.js files
```

Record pinned versions in `manifest.json` description field.

## Step 3 — Data pipeline

```bash
coder data ingest "material-ui/packages/mui-material/src/**/*.tsx" --output mui-raw.jsonl
coder data ingest "module-federation-examples/**/*.{ts,tsx,js}" --output mfe-raw.jsonl

coder data extract --adaptor react-ts --input mui-raw.jsonl --output mui-extracted.jsonl
coder data extract --adaptor react-ts --input mfe-raw.jsonl --output mfe-extracted.jsonl

cat mui-extracted.jsonl mfe-extracted.jsonl > combined.jsonl
coder data deduplicate combined.jsonl --output deduped.jsonl
coder data validate deduped.jsonl --output validated.jsonl
coder data stats validated.jsonl
coder data split validated.jsonl --output-dir ~/.coder/adaptors/react-ts/data/
```

Target: 500–1000 records after deduplication and validation.

## Step 4 — `extract.json`

```json
{
  "rules": [
    { "prompt": "jsdoc",        "completion": "next_function"    },
    { "prompt": "line_comment", "completion": "next_block"       },
    { "prompt": "ts_declare",   "completion": "declare_body"     },
    { "prompt": "line_comment", "completion": "constructor_call" }
  ]
}
```

## Step 5 — Baseline experiment (run BEFORE training)

Before training, establish a baseline score on the eval split using the **base model with no adaptor**. This is the control that proves the adaptor adds measurable value.

```bash
# Score base model (no adaptor) against eval prompts
coder adaptor eval react-ts --baseline

# This runs the eval suite against outputs from:
#   coder generate "<eval prompt>" --model <base-model>
# with NO --adaptor flag, records composite score as baseline_pass_rate in manifest.json
```

Record `baseline_pass_rate` in `manifest.json`:

```json
{
  "baseline_pass_rate": 0.0,   ← filled in before training
  "eval_pass_rate": 0.0        ← filled in after training
}
```

**The adaptor is considered successful if `eval_pass_rate - baseline_pass_rate >= 0.15`** (15 percentage point lift). If the lift is smaller, inspect which dimensions improved and which did not, and retrain with adjusted data or system prompt.

## Step 6 — Training

```bash
coder adaptor train --config react-ts/train-config.toml
```

```toml
[model]
path = "~/.coder/models/mlx-community/Qwen2.5-Coder-7B-Instruct-4bit"

[lora]
rank = 8
target_modules = ["q_proj", "v_proj"]
iters = 1000
batch_size = 4
learning_rate = 1e-4

[data]
dir = "~/.coder/adaptors/react-ts/data"

[output]
adaptor_dir = "~/.coder/adaptors/react-ts/weights"
manifest = "~/.coder/adaptors/react-ts/manifest.json"
log_file = "~/.coder/adaptors/react-ts/training.log"
```

Expected duration: 30–60 min on M3. Monitor loss curve — should decrease steadily.

## Step 7 — System prompt (`prompts/system.md`)

Write a system prompt that instructs the model on:
- React functional components with TypeScript prop interfaces
- MUI component usage (`sx` prop, theme, component variants)
- Module Federation: `exposes`, `remotes`, `shared` config patterns
- TypeScript `declare module` contracts for remote components
- Host/remote app structure conventions

## Step 8 — Eval suite (`evals/eval_suite.ts`)

Write ≥10 test cases covering:
- [ ] MUI Button with typed props
- [ ] MUI TextField with validation
- [ ] `ModuleFederationPlugin` config has required fields (`name`, `exposes`, `shared`)
- [ ] Remote component type declaration is valid TypeScript
- [ ] Shared MUI theme provider wraps host app correctly
- [ ] Custom hook with MUI state follows naming conventions
- [ ] MFE remote exposes at least one component
- [ ] Host app lazy-loads remote component correctly
- [ ] Generated component has no `any` types
- [ ] ESLint passes on generated output

## Step 9 — Run eval, compare, iterate

```bash
coder adaptor eval react-ts
```

Compare `eval_pass_rate` vs `baseline_pass_rate`. Target: ≥0.15 lift, with `eval_pass_rate ≥ 0.70`.

If lift is insufficient:
1. Inspect which scoring dimensions improved (tsc, eslint, test pass rate)
2. Add targeted training examples for failing categories
3. Retrain with `--resume` (checkpoint resumption is automatic)
4. Re-eval and compare

## Step 10 — Package and publish

```bash
cd ~/.coder/adaptors/react-ts
git init && git add . && git commit -m "react-ts adaptor v1.0.0"
coder adaptor install react-ts --from-git <url>
```

## Deliverables

```
adaptors/react-ts/
├── weights/adaptor.safetensors    # trained LoRA weights
├── data/train.jsonl               # ~450–900 training records
├── data/valid.jsonl               # ~50–100 eval records
├── evals/eval_suite.ts            # ≥10 test cases
├── prompts/system.md              # system prompt
├── extract.json                   # extraction rules
└── manifest.json                  # version, baseline_pass_rate, eval_pass_rate ≥ 0.70
```

## Acceptance criteria

- [ ] Step 1: `ts_declare` and `constructor_call` anchors implemented and tested
- [ ] Step 2: Both source repos cloned and version-pinned
- [ ] Step 3: Data pipeline runs end-to-end, ≥500 validated records
- [ ] Step 4: `extract.json` committed with all four rule types
- [ ] Step 5: `baseline_pass_rate` recorded in `manifest.json` before training
- [ ] Step 6: Training completes, loss curve shows convergence
- [ ] Step 7: System prompt committed
- [ ] Step 8: Eval suite with ≥10 test cases committed
- [ ] Step 9: `eval_pass_rate ≥ 0.70` AND lift ≥ 0.15 over baseline
- [ ] Step 10: Adaptor installable via `--from-git`

## Notes
- Blocks #12 (GraphQL adaptor)
- `baseline_pass_rate` must be recorded before training — do not skip this step
- If data yield from extraction is too low (<300 records), supplement with manually authored examples
- `baseline_pass_rate` field needs to be added to the Zod manifest schema before this issue starts
